app/src/main/java/com/x8bit/bitwarden/MainViewModel.kt

```diff
@@ -234,7 +238,20 @@ class MainViewModel @Inject constructor(
         val autofillSaveItem = intent.getAutofillSaveItemOrNull()
         val autofillSelectionData = intent.getAutofillSelectionDataOrNull()
         val shareData = intentManager.getShareDataFromIntent(intent)
-        val totpData = intent.getTotpDataOrNull()
+        val totpData: TotpData? =
+            // First grab TOTP URI directly from the intent data:
+            intent.getTotpDataOrNull()
+                ?: run {
+                    // Then check to see if the intent is coming from the Authenticator app:
+                    if (intent.isAddTotpLoginItemFromAuthenticator()) {
+                        addTotpItemFromAuthenticatorManager.pendingAddTotpLoginItemData.also {
+                            // Clear pending add TOTP data so it is only handled once:
+                            addTotpItemFromAuthenticatorManager.pendingAddTotpLoginItemData = null
```

Comment:
I talked with Dizzy a bit about potentially pulling this logic into a separate extension, but we agreed it was a bit awkward to pull out because of the need to clear `pendingAddTotpLoginItemData`. Definitely open to ideas about cleaning up or isolating this bit.